osquery/tables/system/linux/system_info.cpp

```diff
@@ -40,15 +41,12 @@ QueryData genSystemInfo(QueryContext& context) {
     }
   }
 
-  // Can parse /proc/cpuinfo or /proc/meminfo for this data.
-  static long cores = sysconf(_SC_NPROCESSORS_CONF);
-  if (cores > 0) {
-    r["cpu_logical_cores"] = INTEGER(cores);
-    r["cpu_physical_cores"] = INTEGER(cores);
-  } else {
-    r["cpu_logical_cores"] = "-1";
-    r["cpu_physical_cores"] = "-1";
-  }
+  auto logical_cores = boost::thread::hardware_concurrency();
+  r["cpu_logical_cores"] = (logical_cores > 0) ? INTEGER(logical_cores) : "-1";
+
+  auto physical_cores = boost::thread::physical_concurrency();
+  r["cpu_physical_cores"] =
```

Comment:
According to boost docs this should never return 0.

Reply:
Ah yeah.  I didn't see the docs say it can return `0`, but for some reason I presumptuously assumed that was the intent.   Will fix, thanks for catching that.